### PR TITLE
BYOC update e2e

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/3316

### CODE REVIEW

File: integration/template/azure_operator_charts_values.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ var AzureOperatorChartValues = `Installation:
           CIDR: "10.0.0.0/16"
           ResourceGroup: "godsmack"
           VirtualNetwork: "godsmack"
+          VirtualNetworkGateway: "GatewayTest"
```
Comment (TheoBrigitte) [inline integration/template/azure_operator_charts_values.go]: TODO: align with godsmack vpn gateway

File: helm/azure-operator-resource-chart/templates/cluster.yaml [hunk 2]
```diff
@@ -25,6 +25,7 @@ spec:
       cidr: "{{ .Values.azure.cidr }}"
       masterSubnetCIDR: "{{ .Values.azure.masterSubnetCIDR }}"
       workerSubnetCIDR: "{{ .Values.azure.workerSubnetCIDR }}"
+      vpnSubnetCidr: "{{ .Values.azure.vpnSubnetCidr }}"
```
Comment (kopiczko) [inline helm/azure-operator-resource-chart/templates/cluster.yaml]: I think we should just remove `examples` directory and `helm/azure-operator-resource-chart` directory.

File: integration/setup/setup.go [hunk 3]
```diff
@@ -92,3 +105,45 @@ func Resources(c *client.AzureClientSet, g *framework.Guest, h *framework.Host)
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name: credentialName,
+			},
+
+			// TODO: change to guest subscription credential.
+			Data: map[string][]byte{
+				credential.ClientIDKey:       []byte(c.ClientID),
+				credential.ClientSecretKey:   []byte(c.ClientSecret),
+				credential.SubscriptionIDKey: []byte(c.SubscriptionID),
+				credential.TenantIDKey:       []byte(c.TenantID),
```
Comment (kopiczko) [inline integration/setup/setup.go]: I get why they are exported now.

File: integration/setup/setup.go [hunk 4]
```diff
@@ -92,3 +105,45 @@ func Resources(c *client.AzureClientSet, g *framework.Guest, h *framework.Host)
...
 	return nil
 }
+
+func installCredential(h *framework.Host) error {
+	o := func() error {
+		k8sClient := h.K8sClient()
+
+		k8sClient.CoreV1().Secrets(credentialNamespace).Delete(credentialName, &metav1.DeleteOptions{})
+
+		c := env.AzureGuestConfig()
```
Comment (kopiczko) [inline integration/setup/setup.go]: Why do we need to create a `client.AzureClientSetConfig ` here? I think this is confusing. It suggests creating a client. I'd just take raw values for the `Data map[string][]byte` from `os.Getenv`.

File: helm/azure-operator-resource-chart/templates/cluster.yaml (hunk 2 above)
Comment (TheoBrigitte) [reply helm/azure-operator-resource-chart/templates/cluster.yaml]: Some cleaning would be needed there, let's do that in a different PR.

File: integration/template/azure_operator_charts_values.go (hunk 1 above)
Comment (TheoBrigitte) [reply integration/template/azure_operator_charts_values.go]: @r7vme do you know how will the host cluster VPN Gateway will be named ?

Comment (r7vme) [reply integration/template/azure_operator_charts_values.go]: https://github.com/giantswarm/giantnetes-terraform/pull/111/files#diff-22469f75baefa6503494b40c9b8b860bR11

godsmack-vpn-gateway

**Comment by TheoBrigitte** [comment]
all good?

File: integration/setup/setup.go [hunk 5]
```diff
@@ -92,3 +105,50 @@ func Resources(c *client.AzureClientSet, g *framework.Guest, h *framework.Host)
...
+		k8sClient := h.K8sClient()
+
+		k8sClient.CoreV1().Secrets(credentialNamespace).Delete(credentialName, &metav1.DeleteOptions{})
+
+		c := client.AzureClientSetConfig{
+			ClientID:       env.AzureGuestClientID(),
+			ClientSecret:   env.AzureGuestClientSecret(),
+			SubscriptionID: env.AzureGuestSubscriptionID(),
+			TenantID:       env.AzureGuestTenantID(),
+		}
```
Comment (kopiczko) [inline integration/setup/setup.go]: I just don't follow why should we use `AzureClientSetConfig` here.

You could just do:

```
+				credential.ClientIDKey:       []byte(env.AzureGuestClientID()),
+				credential.ClientSecretKey:   []byte(env.AzureGuestClientSecret()),
+				credential.SubscriptionIDKey: []byte(env.AzureGuestSubscriptionID()),
+				credential.TenantIDKey:       []byte(env.AzureGuestTenantID()),
```

But being in scope of a single method it's now easier to realise that `AzureClientSetConfig` isn't use to create actual client, so it isn't a blocker.

Comment (TheoBrigitte) [reply integration/setup/setup.go]: you are right, no need for that. removing this

**Comment by TheoBrigitte** [comment]
Re-opening as this was merged to the wrong branch

**Pull request merged**